aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -106,18 +105,21 @@ impl CanonicalRequest {
             ..Default::default()
         };
 
-        if let Some(path) = req.uri().query() {
-            let params: BTreeMap<String, String> = qs::from_str(path)?;
-            let n = params.len();
+        if let Some(query) = req.uri().query() {
+            let mut first = true;
             let mut out = String::new();
-            for (i, (k, v)) in params.into_iter().enumerate() {
-                let last = i == n - 1;
-                out.push_str(&percent_encode(&k));
-                out.push('=');
-                out.push_str(&percent_encode(&v));
-                if !last {
+            let mut params: Vec<(Cow<str>, Cow<str>)> =
+                form_urlencoded::parse(query.as_bytes()).collect();
+            params.sort_by(|a, b| a.0.cmp(&b.0));
```

Comment:
nit: use `sort_by_key(|(k,v)|k)` to be a little clearer what's going on. But also, do we not want to resolve the sort by value? what if there are duplicate keys? is it supposed to be a stable sort?